src/pluginlaunch.c

```diff
@@ -119,8 +119,19 @@ update_running_processes (kb_t kb)
                            "ERRMSG|||%s||| |||general/tcp|||%s|||"
                            "NVT timed out after %d seconds.",
                            hostname, oid ?: " ", processes[i].timeout);
+                  kb_item_push_str (main_kb, "internal/results", msg);
 
-                  kb_item_push_str (kb, "internal/results", msg);
+                  /* Check if host is still alive and send a message
+                     if it is dead. */
+                  if (check_host_still_alive (kb, hostname) == 0)
```

Comment:
I would not check the alive status every time we run into a NVT timeout. I think it would be better if we only do the alive check if the maximum number of allowed NVT timeouts were reached. We currently only set the host to dead if the max number is reached, so it does not make sense to check for alive status every time.

Maybe add another functions like `check_max_nvt_timouts` and move all the counter logic into that. Then let this function call the stripped down `check_host_still_alive` which then only checks the alive status if the specified max was reached.